File: 6/robot_return_to_origin.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        l, r, u, d = 0, 0, 0, 0
        for c in moves:
            if c == "L":
                l += 1
            if c == "R":
                r += 1
            if c == "U":
                u += 1
            if c == "D":
                d += 1
        return l==r and u==d

File: 6/robot_return_to_origin.go
package main

import (
	"fmt"
)

func judgeCircle(moves string) bool {
	var l, r, u, d int
	for _, c := range moves {
		switch c {
		case 'L':
			l++
		case 'R':
			r++
		case 'U':
			u++
		case 'D':
			d++
		}
	}
	return l == r && u == d
}

func main() {
	fmt.Println(judgeCircle("LLRR"))
	fmt.Println(judgeCircle("LLR"))
}
